### PR TITLE
Shard NL integration tests to speed up PR test

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: run_test.sh [hunk 1]
```diff
@@ -233,9 +233,15 @@ while [[ "$#" -gt 0 ]]; do
         shift 1
         ;;
     --explore)
-        echo --explore "### Running explore page integration tests"
-        run_integration_test explore_test.py
-        shift 1
+        if [[ -n "$2" && "$2" != -* ]]; then
```
Comment (pradh) [inline run_test.sh:236]: what does `-*` mean? :)

File: run_test.sh [hunk 2]
```diff
@@ -173,7 +173,7 @@ function run_integration_test {
   export TEST_MODE=test
   export ENABLE_EVAL_TOOL=false
 
-  python3 -m pytest -vv --reruns 2 server/integration_tests/$1
+  python3 -m pytest -vv --reruns 2 server/integration_tests/$1 -k $2
```
Comment (pradh) [inline run_test.sh:176]: comment about `-k $2` ?

**Review by pradh — APPROVED** [review]
Very nice!!

File: run_test.sh (hunk 2 above)
Comment (shifucun) [reply run_test.sh]: Done

File: run_test.sh (hunk 1 above)
Comment (shifucun) [reply run_test.sh:243]: this is to check the argument is not like "-p" as an option (from llm..)

**Review by shifucun — COMMENTED** [review]
thanks!

**Pull request merged**